gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -176,61 +165,49 @@ def credentials_present?
 
     def assume_role_from_profile(cfg, profile, opts, chain_config)
```

Comment:
If you want a challenge..... this method could be similar to our credential provider chain rather than a long `if-else`